File: sdk/node/src/api/learning.path.courses/learning.path.courses.ts

import { LearningPathCoursesCreateModel, LearningPathCoursesUpdateModel, LearningPathCoursesSearchParams, ApiResponse } from './learning.path.courses.model';
import { APIClient } from '../client';
import { plainToInstance } from 'class-transformer';
import { validateSync } from 'class-validator';
import { faker } from '@faker-js/faker';

export class LearningPathCourses {
    constructor(private apiClient: APIClient) {}

    private validateDataclass<T extends object>(data: Record<string, any>, DataClass: new () => T): T 
    {
        const instance = plainToInstance(DataClass, data as Partial<T>); 
        const errors = validateSync(instance);
        if (errors.length > 0) {
            throw new Error(`Validation failed: ${JSON.stringify(errors)}`);
        }
        return instance;
    }

    public async create(data: Record<string, any>): Promise<ApiResponse<LearningPathCoursesCreateModel>> 
    {
        const validatedData = this.validateDataclass(data, LearningPathCoursesCreateModel);
        return this.apiClient.makeRequest<ApiResponse<LearningPathCoursesCreateModel>>('POST', 'learning_path_courses', validatedData);
    }

    public async getById(learning_path_courses_Id: string): Promise<ApiResponse<LearningPathCoursesCreateModel>> 
    {
        if (typeof learning_path_courses_Id !== 'string') {
            throw new TypeError(`Expected 'learning_path_courses_Id' to be a string, got ${typeof learning_path_courses_Id}.`);
        }
        return this.apiClient.makeRequest<ApiResponse<LearningPathCoursesCreateModel>>('GET', `learning_path_courses/${ learning_path_courses_Id }`);
    }

    public async search(queryParams: Record<string, any>): Promise<ApiResponse<LearningPathCoursesCreateModel[]>> 
    {
        const validatedData = this.validateDataclass(queryParams, LearningPathCoursesSearchParams);
        return this.apiClient.makeRequest<ApiResponse<LearningPathCoursesCreateModel[]>>('GET', `learning_path_courses/search`, validatedData);
    }

    public async update(learning_path_courses_Id: string, data: Record<string, any>): Promise<ApiResponse<LearningPathCoursesCreateModel>> 
    {
        if (typeof learning_path_courses_Id !== 'string') {
            throw new TypeError(`Expected 'learning_path_courses_Id' to be a string, got ${typeof learning_path_courses_Id}.`);
        }
        const validatedData = this.validateDataclass(data, LearningPathCoursesUpdateModel);
        return this.apiClient.makeRequest<ApiResponse<LearningPathCoursesCreateModel>>('PUT', `learning_path_courses/${ learning_path_courses_Id }`, validatedData);
    }

    public async delete(learning_path_courses_Id: string): Promise<ApiResponse<void>> 
    {
        if (typeof learning_path_courses_Id !== 'string') {
            throw new TypeError(`Expected 'learning_path_courses_Id' to be a string, got ${typeof learning_path_courses_Id}.`);
        }
        return this.apiClient.makeRequest<ApiResponse<void>>('DELETE', `learning_path_courses/${ learning_path_courses_Id }`);
    }

    public async learningPathCoursesOperations(): Promise<void> 
    {
        try {
            const learningPathCoursesCreateData = {

            };

            const learningPathCoursesCreate = await this.create(learningPathCoursesCreateData);
            console.log('Create:', JSON.stringify(learningPathCoursesCreate.Data, null, 2));

            const learning_path_courses_Id = learningPathCoursesCreate.Data?.id; 
            if (!learning_path_courses_Id) throw new Error('LearningPathCourses ID not returned.');

            const learningPathCoursesGetById = await this.getById(learning_path_courses_Id);
            console.log('GetById:', JSON.stringify(learningPathCoursesGetById.Data, null, 2));

            const learningPathCoursesSearchData = {

            };
            console.log('Search Params:', learningPathCoursesSearchData);

            const learningPathCoursesSearch = await this.search(learningPathCoursesSearchData);
            console.log('Search:', JSON.stringify(learningPathCoursesSearch.Data, null, 2));

            const learningPathCoursesUpdateData = {

            };

            const learningPathCoursesUpdate = await this.update(learning_path_courses_Id, learningPathCoursesUpdateData);
            console.log('Update:', JSON.stringify(learningPathCoursesUpdate.Data, null, 2));

            const learningPathCoursesDelete = await this.delete(learning_path_courses_Id);
            console.log('Delete:', JSON.stringify(learningPathCoursesDelete.Data, null, 2));

        } catch (error) {
            console.error('Error during learning_path_courses operations:', error);
        }
    }
}

File: sdk/python/src/lmsservice_db/api/learning_path_courses/learning_path_courses.py

import json
from dataclasses import asdict
from typing import Dict
from faker import Faker
from lmsservice_db.api.learning_path_courses.learning_path_courses_model import LearningPathCoursesCreateModel, LearningPathCoursesSearchParams, LearningPathCoursesUpdateModel

class LearningPathCourses:
    def __init__(self, client):
        self.client = client
        self.faker = Faker()

    def validate_dataclass(self, data: Dict, dataclass_type):
        """
        Validates the input data against the specified dataclass type.
        """
        try:
            instance = dataclass_type(**data)
            return instance
        except TypeError as e:
            raise ValueError(f"Validation failed: {e}")

    def create_learning_path_courses(self, data: Dict):
        validated_data = self.validate_dataclass(data, LearningPathCoursesCreateModel)
        endpoint = "learning_path_courses"
        return self.client.make_request('POST', endpoint, data=asdict(validated_data))

    def get_learning_path_courses_by_id(self, learning_path_courses_id: str):
        if not isinstance(learning_path_courses_id, str):
            raise TypeError(f"Expected 'learning_path_courses_id' to be a string, got {type(learning_path_courses_id).__name__}.")
        endpoint = f'learning_path_courses/{learning_path_courses_id}'
        return self.client.make_request('GET', endpoint)

    def search_learning_path_courses(self, query_params: Dict):
        validated_data = self.validate_dataclass(query_params, LearningPathCoursesSearchParams)
        endpoint = 'learning_path_courses/search'
        return self.client.make_request('GET', endpoint, data=asdict(validated_data))

    def update_learning_path_courses(self, learning_path_courses_id: str, data: Dict):
        if not isinstance(learning_path_courses_id, str):
            raise TypeError(f"Expected 'learning_path_courses_id' to be a string, got {type(learning_path_courses_id).__name__}.")
        validated_data = self.validate_dataclass(data, LearningPathCoursesUpdateModel)
        endpoint = f'learning_path_courses/{learning_path_courses_id}'
        return self.client.make_request('PUT', endpoint, data=asdict(validated_data))

    def delete_learning_path_courses(self, learning_path_courses_id: str):
        if not isinstance(learning_path_courses_id, str):
            raise TypeError(f"Expected 'learning_path_courses_id' to be a string, got {type(learning_path_courses_id).__name__}.")
        endpoint = f'learning_path_courses/{learning_path_courses_id}'
        return self.client.make_request('DELETE', endpoint)

    @staticmethod
    async def execute_learning_path_courses_operations(client):
        faker = Faker()

        learning_path_courses_create_data = {
            
        }

        learning_path_courses_update_data = {
            
        }

        search_params = {
            
        }

        new_user = client.learning_path_courses.create_learning_path_courses(learning_path_courses_create_data)
        print('Create:', json.dumps(new_user, indent=2))
        print("\n")

        learning_path_courses_id = new_user['Data']['id']
        user = client.learning_path_courses.get_learning_path_courses_by_id(learning_path_courses_id)
        print('GetById:', json.dumps(user, indent=2))
        print("\n")

        search_results = client.learning_path_courses.search_learning_path_courses(search_params)
        print('Search:', json.dumps(search_results, indent=2))
        print("\n")

        updated_user = client.learning_path_courses.update_learning_path_courses(learning_path_courses_id, learning_path_courses_update_data)
        print('Update:', json.dumps(updated_user, indent=2))
        print("\n")

        deleted_user = client.learning_path_courses.delete_learning_path_courses(learning_path_courses_id)
        print('Delete:', json.dumps(deleted_user, indent=2))
        print("\n")

